w1r3/go/main.go

```diff
@@ -0,0 +1,582 @@
+	log.Printf("# Go SDK Version: %s", versions["cloud.google.com/go/storage"])
+	log.Printf("# gRPC Version: %s", versions["google.golang.org/grpc"])
+	log.Printf("# Protobuf Version: %s", versions["google.golang.org/protobuf"])
+	log.Printf("# Tracing Rate: %f", *tracingRate)
+	log.Printf("# Version for Profiler: %s", *profileVersion)
+
+	config := Config{
+		transports:  transports,
+		uploaders:   uploaders,
+		objectSizes: objectSizes,
+		bucketName:  *bucket,
+		deployment:  *deployment,
+		instance:    instance.String(),
+		versions:    versions,
+		iterations:  *iterations,
+	}
+
+	var wg sync.WaitGroup
+	launch := func() {
+		defer wg.Done()
+		worker(ctx, config)
+	}
+
+	wg.Add(*workers)
+	for range *workers {
+		go launch()
+	}
+	wg.Wait()
+}
+
+type Config struct {
+	transports  []Transport
+	uploaders   []Uploader
+	objectSizes intFlags
+	bucketName  string
+	deployment  string
+	instance    string
+	versions    map[string]string
+	iterations  int
+}
+
+func worker(ctx context.Context, config Config) {
+	tracer := otel.GetTracerProvider().Tracer("")
+	meter := otel.GetMeterProvider().Meter("")
+	histogram, err := meter.Int64Histogram(
+		"ssb/w1r3/latency",
+		metric.WithDescription("The duration of task execution."),
+		metric.WithUnit("ms"),
+		metric.WithExplicitBucketBoundaries(histogramBoundaries()...),
+	)
+	if err != nil {
+		log.Fatalf("Cannot create ssb/w1r3/latency histogram")
+	}
+	get_version := func(name string) string {
+		v, ok := config.versions[name]
+		if ok {
+			return v
+		}
+		return "unknown"
+	}
+
+	data := make([]byte, slices.Max(config.objectSizes))
+	rand.Read(data) // rand.Read() is deprecated, but good enough for this benchmark.
+	for i := range config.iterations {
+		id, err := uuid.NewRandom()
+		if err != nil {
+			continue
+		}
+		var objectName = id.String()
+		var objectSize = config.objectSizes[rand.Intn(len(config.objectSizes))]
+		var transport = config.transports[rand.Intn(len(config.transports))]
+		if transport.client == nil {
+			log.Fatalf("Iteration %d transport %s is nil", i, transport.name)
+		}
+		var uploader = config.uploaders[rand.Intn(len(config.uploaders))]
+
+		commonAttributes := []attribute.KeyValue{
+			attribute.String("ssb.language", "go"),
+			attribute.Int64("ssb.object-size", objectSize),
+			attribute.String("ssb.transport", transport.name),
+			attribute.String("ssb.deployment", config.deployment),
+			attribute.String("ssb.instance", config.instance),
+			attribute.String("ssb.version", benchmarkVersion()),
+			attribute.String("ssb.version.sdk", get_version("cloud.google.com/go/storage")),
+			attribute.String("ssb.version.grpc", get_version("google.golang.org/grpc")),
+			attribute.String("ssb.version.protobuf", get_version("google.golang.org/protobuf")),
+			attribute.String("ssb.version.http", get_version("golang.org/x/net")),
+		}
+
+		spanContext, span := tracer.Start(
+			ctx, "ssb::iteration", trace.WithAttributes(
```

Comment:
I am not sure what "client" you mean.  The storage SDK client?  No.  The gRPC and HTTP clients?  Yes.  If there is a way to enable OTel spans for the storage SDK I would lose to turn those on.

Reply:
Yes, there is a "bridge" that must be added until the (stoage SDK) client is fully migrated from OC to OT. See https://github.com/googleapis/google-cloud-go/blob/main/debug.md#opencensus . In a couple months it will no longer be needed though. It's basically just adding these two lines: https://github.com/open-telemetry/opentelemetry-go/blob/bridge/opencensus/examples/bridge/v0.14.0/bridge/opencensus/examples/bridge/main.go#L44

Reply:
Thanks, following up in #41